libcudacxx/include/cuda/std/detail/libcxx/include/__cuda/ptx/parallel_synchronization_and_communication_instructions_mbarrier.h

```diff
@@ -534,10 +534,11 @@ _LIBCUDACXX_DEVICE static inline bool mbarrier_test_wait(
         "l"(__state)
       : "memory"
     );
-    return static_cast<bool>(__waitComplete); // Deliberate downcast.
+    return static_cast<bool>(__waitComplete);
   ),(
     // Unsupported architectures will have a linker error with a semi-decent error message
-    return __bool__cuda_ptx_mbarrier_test_wait_is_not_supported_before_SM_80__();
+    __cuda_ptx_mbarrier_test_wait_is_not_supported_before_SM_80__();
+    return static_cast<bool>(0);
```

Comment:
Important: I would prefer if we would just return `false` here and below
```suggestion
    return false;
```

Reply:
You are right. I have changed the code to do this now.